packet/https.go

```diff
@@ -42,8 +43,12 @@ func ReadTLSMessage(r io.Reader) (*TLSMessage, error) {
 		ProtoVersion: binary.BigEndian.Uint16(rawHeader[1:3]),
 		PayloadLen:   binary.BigEndian.Uint16(rawHeader[3:5]),
 	}
-
-	raw := make([]byte, header.PayloadLen+TLSHeaderLen)
+	rawLen := header.PayloadLen + TLSHeaderLen
+	if rawLen < TLSHeaderLen {
```

Comment:
It's worth thinking about the maximum payload. I think 8192 will be enough, I haven't seen more in practice